pkg/docker/runner.go

```diff
@@ -101,7 +101,11 @@ func (n *Runner) Run(ctx context.Context, f fn.Function) (job *fn.Job, err error
 		}
 	}()
 
-	// Start
+	// TODO: use StartTimeout
```

Comment:
Presumably this is why it's a work in progress? 😄

Reply:
Haha, no no.  The flag --start-timeout was a necessary digression in order to get the host-based runner functioning.  So this PR only implements that path:  a host-based (non-containerized) run.   It might be two good first issues to have the flag also respected for containerized runs, and by the knative deployer.